### ISSUE TITLE
[FEATURE] Add retry for BWC tests

### ISSUE THREAD

**Issue by chishui** [opensearch-project/neural-search#778]
### Is your feature request related to a problem?
BWC tests are quite flaky, here is one failed test example:
- https://github.com/opensearch-project/neural-search/actions/runs/9377650430/job/25871682884

We can introduce retry for BWC tests to reduce the chance of random failures. 

### What solution would you like?
OpenSearch core repo uses test-retry plugin to retry flaky tests: https://github.com/opensearch-project/OpenSearch/blob/main/build.gradle#L469

### What alternatives have you considered?
N/A

### Do you have any additional context?
N/A

**Comment by dblock** [comment]
Double check that the errors are truly transient. Retrying often just hides eventual consistency errors and is not recommended.

[Catch All Triage - Attendees [1](https://github.com/dblock/), [2](https://github.com/krisfreedain), [3](https://github.com/andrross), [4](https://github.com/gkamat), [5](https://github.com/getsaurabh02)]

**Comment by heemin32** [comment]
I agree that addressing the root cause of test flakiness should take priority over introducing a retry mechanism. Currently, there is one flaky test that we are actively working on fixing. We'll close this for now and handle each flaky test individually rather than implementing retry logic. https://github.com/opensearch-project/neural-search/issues/1035
